includes/Specials/SpecialManageWikiDefaultPermissions.php

```diff
@@ -31,6 +32,10 @@ public function execute( $par ) {
 			return false;
 		}
 
+		if ( MediaWikiServices::getInstance()->getReadOnlyMode()->isReadOnly() ) {
```

Comment:
You can just use checkReadOnly(); here, if you want it to warn instead of error, then you can use a try/catch, catching ReadOnlyError, and output $ex that it returns, which would be exactly equivalent to this, since this is the same it throws. But It can probably just use the exception and error out if readonly.

However, the issue is this doesn't really apply, this needs to somehow check, not local read only, but the LB read only, the one on the global CreateWiki database, as this doesn't use local DB. I don't think this is necessarily needed, because if the global DB is readonly, it'll exit out anyway.